klayout/drc/rule_decks/hres.drc

```diff
@@ -0,0 +1,150 @@
+  hres3_l1 = hres_poly.space(0.4.um, euclidian)
+  hres3_l1.output('HRES.3', 'HRES.3 : Minimum space between Poly2 resistors. : 0.4µm')
+  hres3_l1.forget
+
+  # Rule HRES.4: Minimum RESISTOR overlap of Poly2 resistor. is 0.4µm
+  logger.info('Executing rule HRES.4')
+  hres4_l1 = hres_poly.enclosed(resistor, 0.4.um, euclidian).polygons(0.001.um)
+  hres4_l2 = hres_poly.not_outside(resistor).not(resistor)
+  hres4_l  = hres4_l1.or(hres4_l2)
+  hres4_l.output('HRES.4', 'HRES.4 : Minimum RESISTOR overlap of Poly2 resistor. : 0.4µm')
+  hres4_l1.forget
+  hres4_l2.forget
+  hres4_l.forget
+
+  # Rule HRES.5: Minimum RESISTOR space to unrelated Poly2. is 0.3µm
+  logger.info('Executing rule HRES.5')
+  hres5_l1 = resistor.interacting(hres1_poly).separation(poly2.not_interacting(sab), 0.3.um, euclidian)
+  hres5_l1.output('HRES.5', 'HRES.5 : Minimum RESISTOR space to unrelated Poly2. : 0.3µm')
+  hres5_l1.forget
+
+  # Rule HRES.6: Minimum RESISTOR space to COMP.
+  logger.info('Executing rule HRES.6')
+  hres6_l1 = resistor.interacting(hres1_poly).separation(comp, 0.3.um, euclidian).polygons(0.001.um)
+  hres6_l2 = comp.not_outside(resistor.interacting(hres1_poly))
+  hres6_l = hres6_l1.or(hres6_l2)
+  hres6_l.output('HRES.6', 'HRES.6 : Minimum RESISTOR space to COMP.')
+  hres6_l.forget
+  hres6_l1.forget
+  hres6_l2.forget
+  hres1_poly.forget
+
+  # Rule HRES.7: Minimum Pplus overlap of contact on Poly2 resistor. is 0.2µm
+  logger.info('Executing rule HRES.7')
+  hres7_l1 = contact.and(hres_poly).enclosed(pplus, 0.2.um, euclidian).polygons(0.001.um)
+  hres7_l2 = contact.and(hres_poly).not_outside(pplus).not(pplus)
+  hres7_l  = hres7_l1.or(hres7_l2)
+  hres7_l.output('HRES.7', 'HRES.7 : Minimum Pplus overlap of contact on Poly2 resistor. : 0.2µm')
+  hres7_l1.forget
+  hres7_l2.forget
+  hres7_l.forget
+
+  # Rule HRES.8: Space from salicide block to contact on Poly2 resistor.
+  logger.info('Executing rule HRES.8')
+  hres8_l1 = contact.and(hres_poly).separation(sab,
+                                               0.22.um).polygons(0.001.um).or(contact.and(hres_poly).interacting(sab))
+  hres8_l1.output('HRES.8', 'HRES.8 : Space from salicide block to contact on Poly2 resistor.')
+  hres8_l1.forget
+
+  # Rule HRES.9: Minimum salicide block overlap of Poly2 resistor in width direction.
+  logger.info('Executing rule HRES.9')
+  hres9_sab             = sab.interacting(pplus).interacting(res_mk).interacting(resistor)
+  hres9_clear_sab       = hres9_sab.not(hres_poly)
+  hres9_bad_inside_edge = hres9_sab.edges.inside_part(hres_poly).extended(0, 0, 0.001.um, 0.001.um).interacting(
+    hres9_clear_sab, 1, 1
+  )
+  hres9_sab_hole = hres9_sab.holes.and(hres_poly)
+  hres9_l1 = hres_poly.enclosed(hres9_sab, 0.28.um, euclidian).polygons(0.001.um)
+  hres9_l2 = hres9_bad_inside_edge.or(hres9_sab_hole)
+  hres9_l = hres9_l1.or(hres9_l2)
+  hres9_l.output('HRES.9', 'HRES.9 : Minimum salicide block overlap of Poly2 resistor in width direction.')
```

Comment:
should we add the spacing requirement to the comment (like we do in other rules?)

Reply:
Done